boards/nxp/mimxrt1180_evk/board.cmake

```diff
@@ -0,0 +1,23 @@
+#
+# Copyright 2024, NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(RT1180_BOARD_DIR
+"${ZEPHYR_HAL_NXP_MODULE_DIR}/mcux/mcux-sdk/boards/evkmimxrt1180")
+# Note1: Suggest developers use Secure Provisioning Tool(SPT) to download RT1180 image
+# SPT can be downloaded on NXP web: https://www.nxp.com/design/design-center/software/development-software/mcuxpresso-software-and-tools-/mcuxpresso-secure-provisioning-tool:MCUXPRESSO-SECURE-PROVISIONING
+# Details about the usage of SPT on MIMXRT1180-EVK board can be referred on chapter 7 of getting start with Mcuxpresso SDK for MIMXRT1180-EVK doc in SDK package.
+if(CONFIG_SOC_MIMXRT1189_CM33)
+board_runner_args(linkserver  "--device=MIMXRT1189xxxxx:MIMXRT1180-EVK")
+board_runner_args(jlink "--device=MIMXRT1189xxx8_M33" "--reset-after-load" "--jlinkscriptfile=${RT1180_BOARD_DIR}/jlinkscript/evkmimxrt1180_cm33.jlinkscript")
```

Comment:
cat /home/suram/nxp/mimxrt1180_evk_cm33/temp/mimxrt1180_evk_mimxrt1189_cm33_twister_test_only/twister-out/mimxrt1180_evk_mimxrt1189_cm33/samples/application_development/external_lib/sample.app_dev.external_lib/device.log
FATAL ERROR: runner jlink received unknown arguments: ['--jlinkscriptfile=/home/jenkins/agent/workspace/zephyr_test_pipe/modules/hal/nxp/mcux/mcux-sdk/boards/evkmimxrt1180/jlinkscript/evkmimxrt1180_cm33.jlinkscript']

Reply:
danieldegrasse suggest me to split jlink.py to another PR. So the issue is caused by here.
New PR created：https://github.com/zephyrproject-rtos/zephyr/pull/74226

Reply:
It appears there are outstanding comments on that PR. Is is possible to use `--tool-opt` as was suggested there?

Reply:
Sure, I am working for resolving the timeout issue. Once resolved, I will try to use `--tool-opt` , if ok, I will push together.

Reply:
I have modified to use '--tool-opt' to add script for cm7 core.